server/src/grpc/server.rs

```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
-        self.region
-            .request_waypoints(self.waypoints.clone(), request)
+async fn update_waypoints(host: String, port: u16, waypoints: &HashMap<String, Coordinates>) {
+    let nodes: Vec<svc_gis_client_grpc::client::Waypoint> = waypoints
+        .iter()
+        .map(
+            |(label, coordinates)| svc_gis_client_grpc::client::Waypoint {
+                label: label.clone(),
+                location: Some(*coordinates),
+            },
+        )
+        .collect();
+
+    if nodes.is_empty() {
+        grpc_warn!("(waypoints_loop) no waypoints to update.");
+        return;
     }
 
-    async fn request_restrictions(
-        &self,
-        request: Request<RestrictionsRequest>,
-    ) -> Result<Response<RestrictionsResponse>, Status> {
-        grpc_warn!(
-            "([{}] request_restrictions) compliance server.",
-            self.region.get_region()
-        );
-        grpc_debug!("(request_restrictions) request: {:?}", request);
-        self.region
-            .request_restrictions(self.restrictions.clone(), request)
+    let request = tonic::Request::new(UpdateWaypointsRequest { waypoints: nodes });
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
+        return;
+    };
+
+    match client.update_waypoints(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
+        }
+        Err(e) => {
+            grpc_error!("ERROR={:?}", e);
+        }
+    }
+}
+
+/// Periodically pulls down waypoints from the regional interface and
+///  pushes them to the GIS microservice
+pub async fn waypoints_loop(config: Config, region: Box<dyn RegionInterface + Send + Sync>) {
+    let host = config.gis_host_grpc;
+    let port = config.gis_port_grpc;
+
+    grpc_info!(
+        "(waypoints_loop) Starting loop with interval: {} seconds.",
+        config.interval_seconds_refresh_waypoints
+    );
+
+    let mut cache: HashMap<String, Coordinates> = HashMap::new();
+
+    loop {
+        // Pull down waypoints from regional interface
+        region.acquire_waypoints(&mut cache).await;
+        update_waypoints(host.clone(), port, &cache).await;
+        std::thread::sleep(std::time::Duration::from_secs(
+            config.interval_seconds_refresh_waypoints as u64,
+        ));
+    }
+}
+
+async fn update_restrictions(
+    host: String,
+    port: u16,
+    restrictions: &HashMap<String, RestrictionDetails>,
+) {
+    let mut zones: Vec<NoFlyZone> = vec![];
+    for (label, details) in restrictions.iter() {
+        let time_start = match details.timestamp_start {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
```

Comment:
Wrong function name in log message -> (update_restrictions)

Reply:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1